symphony/bdk/core/retry/strategy.py

```diff
@@ -0,0 +1,120 @@
+from aiohttp import ClientConnectorError
+from tenacity import RetryCallState
+
+from symphony.bdk.core.auth.auth_session import AuthSession
+from symphony.bdk.core.auth.exception import AuthUnauthorizedError
+from symphony.bdk.core.service.datafeed.abstract_datafeed_loop import AbstractDatafeedLoop
+from symphony.bdk.gen import ApiException
+
+
+def is_unauthorized(exception: Exception) -> bool:
+    return isinstance(exception, ApiException) and exception.status == 401
+
+
+def is_client_error(exception: Exception) -> bool:
+    return isinstance(exception, ApiException) and exception.status == 400
+
+
+def is_client_timeout_error(exception: Exception):
+    """Checks if the exception is a :py:class:`ClientConnectorError` with a :py:class:`TimeoutError` as cause
+
+    :param exception: The exception to be checked
+    :return: True if checks the predicate, False otherwise
+    """
+    return isinstance(exception, ClientConnectorError) and isinstance(exception.__cause__, TimeoutError)
+
+
+def can_authentication_be_retried(exception: Exception) -> bool:
+    """Checks if the exception is a network issue or internal or retry error
+    i.e. is an :py:class:`ApiException` with status code 429 or greater or equal to 500.
+    or the exception is either a :py:cass:`TimeoutError`  (e.g. network issues)
+
+    :param exception: The exception to be checked
+    :return: True if the exception is an :py:class:`ApiException` with status 500, 429 or or is a client timeout error
+             False otherwise
+    """
+    if isinstance(exception, ApiException):
+        return exception.status >= 500 or exception.status == 429
+    return is_client_timeout_error(exception)
+
+
+def is_network_or_minor_error(exception: Exception) -> bool:
+    """Checks if the exception is a network issue or an :py:class:`ApiException` minor error (internal + unauthorized + retry)
+    This is the default function used to check if a given exception should lead to a retry
+
+    :param exception: The exception to be checked
+    :return: True if the exception is an :py:class:`ApiException` with status 500, 401, 429 or is a client timeout error
+             False otherwise
+    """
+    if isinstance(exception, ApiException):
+        if exception.status >= 500 or exception.status == 401 or exception.status == 429:
```

Comment:
instead of `if` + `return`: `return exception.status >= 500 or exception.status == 401 or exception.status == 429`